// @shopware-ag/admin-extension-sdk/umd/_internals/utils.js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "lodash/has"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.traverseObject = exports.hasOwnProperty = exports.hasType = exports.getWindowSrc = exports.getLocationId = exports.isObject = exports.generateUniqueId = void 0;
    const has_1 = __importDefault(require("lodash/has"));
    function generateUniqueId() {
        return String(Date.now().toString(36) + Math.random().toString(36).substr(2));
    }
    exports.generateUniqueId = generateUniqueId;
    /* eslint-disable */
    function isObject(value) {
        return value !== null && typeof value === 'object';
    }
    exports.isObject = isObject;
    function getLocationId() {
        const params = new URLSearchParams(window.location.search);
        return params.get('location-id');
    }
    exports.getLocationId = getLocationId;
    function getWindowSrc() {
        const location = window.location;
        const urlObject = new URL(location.pathname, location.origin);
        return urlObject.toString();
    }
    exports.getWindowSrc = getWindowSrc;
    function hasType(type, obj) {
        return isObject(obj) && obj.__type__ && obj.__type__ === type;
    }
    exports.hasType = hasType;
    function hasOwnProperty(obj, path) {
        return (0, has_1.default)(obj, path);
    }
    exports.hasOwnProperty = hasOwnProperty;
    function traverseObject(traversableObject, processor) {
        for (let index in traversableObject) {
            const currentEntry = traversableObject[index];
            processor.apply(this, [traversableObject, index, currentEntry]);
            if (isObject(currentEntry)) {
                traverseObject(currentEntry, processor);
            }
        }
    }
    exports.traverseObject = traverseObject;
});
