// @shopware-ag/admin-extension-sdk/umd/privileges/index.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "./messages/_privileges"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.ShopwareMessageTypePrivileges = void 0;
    const _privileges_1 = require("./messages/_privileges");
    /**
     * Determines which privileges are needed for a certain message type.
     * @internal
     */
    exports.ShopwareMessageTypePrivileges = {
        notificationDispatch: {},
        windowRedirect: {},
        windowReload: {},
        contextLanguage: {},
        contextEnvironment: {},
        contextLocale: {},
        contextCurrency: {},
        contextShopwareVersion: {},
        contextAppInformation: {},
        getPageTitle: {},
        uiComponentSectionRenderer: {},
        uiTabsAddTabItem: {},
        cmsRegisterElement: {},
        locationUpdateHeight: {},
        menuItemAdd: {},
        settingsItemAdd: {},
        mainModuleAdd: {},
        uiModalOpen: {},
        uiModalClose: {},
        actionButtonAdd: {},
        actionExecute: {},
        datasetRegistration: {},
        datasetSubscribe: {},
        datasetUpdate: {},
        __function__: {},
        __registerWindow__: {},
        _criteriaTest: {},
        _collectionTest: {},
        _multiply: {},
        _subtract: {},
        _privileges: _privileges_1._privileges,
    };
});
