// @shopware-ag/admin-extension-sdk/umd/_internals/serializer/index.js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "./function-serializer", "./criteria-serializer", "./entity-serializer", "./entity-collection-serializer", "./handle-error-serializer", "lodash/cloneDeepWith", "./missing-priviliges-error-serializer"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    const function_serializer_1 = __importDefault(require("./function-serializer"));
    const criteria_serializer_1 = __importDefault(require("./criteria-serializer"));
    const entity_serializer_1 = __importDefault(require("./entity-serializer"));
    const entity_collection_serializer_1 = __importDefault(require("./entity-collection-serializer"));
    const handle_error_serializer_1 = __importDefault(require("./handle-error-serializer"));
    const cloneDeepWith_1 = __importDefault(require("lodash/cloneDeepWith"));
    const missing_priviliges_error_serializer_1 = __importDefault(require("./missing-priviliges-error-serializer"));
    /**
     * Collect all single serializer/deserializer. The first matching result
     * will be used as the customizer in cloneDeepWith
     */
    const serializerFactories = [
        criteria_serializer_1.default,
        entity_collection_serializer_1.default,
        entity_serializer_1.default,
        function_serializer_1.default,
        handle_error_serializer_1.default,
        missing_priviliges_error_serializer_1.default,
    ];
    /**
     * The main serializer factory. It returns a general serializer/deserializer which combines
     * all single serializer
     */
    function mainSerializerFactory(dependencies) {
        const serializers = serializerFactories.map(serializerFactory => serializerFactory(dependencies));
        function getSerializers() {
            return serializers;
        }
        function getSerializerByName(name) {
            var _a;
            return (_a = serializers.find(serializer => serializer.name === name)) !== null && _a !== void 0 ? _a : null;
        }
        /* eslint-disable */
        function serialize(messageData) {
            return (0, cloneDeepWith_1.default)(messageData, (value, key, object, stack) => {
                // return first matching serializer result
                for (const serializer of serializers) {
                    const result = serializer.serialize({
                        value,
                        key,
                        object,
                        stack,
                        customizerMethod: serialize,
                    });
                    if (result) {
                        return result;
                    }
                    ;
                }
            });
        }
        function deserialize(messageData, event) {
            return (0, cloneDeepWith_1.default)(messageData, (value, key, object, stack) => {
                // return first matching serializer result
                for (const serializer of serializers) {
                    const result = serializer.deserialize({
                        value,
                        key,
                        object,
                        stack,
                        event,
                        customizerMethod: deserialize,
                    });
                    if (result) {
                        return result;
                    }
                    ;
                }
            });
        }
        /* eslint-enable */
        return {
            getSerializers,
            getSerializerByName,
            serialize,
            deserialize,
        };
    }
    exports.default = mainSerializerFactory;
});
